src/Utilities/Options/AnalyzerOptionsExtensions.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+#pragma warning disable RS1012 // Start action has no registered actions.
+
+namespace Analyzer.Utilities
+{
+    internal static class AnalyzerOptionsExtensions
+    {
+        public static SymbolVisibilityGroup GetSymbolVisibilityGroupOption(
+            this AnalyzerOptions options,
+            DiagnosticDescriptor rule,
+            SymbolVisibilityGroup defaultValue,
+            CancellationToken cancellationToken)
+            => options.GetEnumOptionValue("api_surface", rule, defaultValue, cancellationToken);
+
+        private static TEnum GetEnumOptionValue<TEnum>(
+            this AnalyzerOptions options,
+            string optionName,
+            DiagnosticDescriptor rule,
+            TEnum defaultValue,
+            CancellationToken cancellationToken)
+            where TEnum: struct
+        {
+            var analyzerConfigOptions = options.GetCategorizedAnalyzerConfigOptions(cancellationToken);
+            return analyzerConfigOptions.GetEnumOptionValue(optionName, rule, defaultValue);
+        }
+
+        private static CategorizedAnalyzerConfigOptions GetCategorizedAnalyzerConfigOptions(
+            this AnalyzerOptions options,
+            CancellationToken cancellationToken)
+        {
+            foreach (var additionalFile in options.AdditionalFiles)
```

Comment:
this feels like will be  really expensive. result feels like must be cached. through something like http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs,ba00718202045393

Reply:
since option (and everything under it) is immutable within the same compilation, all data should be only parsed once and cached.

Reply:
Yes, I have cached it here: https://github.com/dotnet/roslyn-analyzers/pull/1952/files/ac97a3e5e8be0d3a7682fe2ea533a4b73ea4905c#diff-9fdf546cd4bb28467e07042b27363bc2R60. I believe once the compiler support comes online, they have to do the caching, reading text, etc., so I have kept current caching logic in the temporary editorconfig parser